lib/tasks/euros.rake

```diff
@@ -0,0 +1,128 @@
+namespace :euros do
+  desc "Create the UEFA Euros competition"
+  task create: :environment do
+    groups = {
+      'Group A' => {
+        api_id: nil,
+        api_code: 'GROUP_A',
+        teams: [
+          { name: 'Germany', abbrev: 'GER' },
+          { name: 'Scotland', abbrev: 'SCO' },
+          { name: 'Switzerland', abbrev: 'SUI' },
+          { name: 'Hungary', abbrev: 'HUN' }
+        ]
+      },
+      'Group B' => {
+        api_id: nil,
+        api_code: 'GROUP_B',
+        teams: [
+          { name: 'Albania', abbrev: 'ALB' },
+          { name: 'Italy', abbrev: 'ITA' },
+          { name: 'Croatia', abbrev: 'CRO' },
+          { name: 'Spain', abbrev: 'ESP' }
+        ]
+      },
+      'Group C' => {
+        api_id: nil,
+        api_code: 'GROUP_C',
+        teams: [
+          { name: 'Denmark', abbrev: 'DEN' },
+          { name: 'England', abbrev: 'ENG' },
+          { name: 'Serbia', abbrev: 'SRB' },
+          { name: 'Slovenia', abbrev: 'SVN' }
+        ]
+      },
+      'Group D' => {
+        api_id: nil,
+        api_code: 'GROUP_D',
+        teams: [
+          { name: 'France', abbrev: 'FRA' },
+          { name: 'Netherlands', abbrev: 'NED' },
+          { name: 'Austria', abbrev: 'AUT' },
+          { name: 'Poland', abbrev: 'POL' }
+        ]
+      },
+      'Group E' => {
+        api_id: nil,
+        api_code: 'GROUP_E',
+        teams: [
+          { name: 'Belgium', abbrev: 'BEL' },
+          { name: 'Romania', abbrev: 'ROU' },
+          { name: 'Slovakia', abbrev: 'SVK' },
+          { name: 'Ukraine', abbrev: 'UKR' }
+        ]
+      },
+      'Group F' => {
+        api_id: nil,
+        api_code: 'GROUP_F',
+        teams: [
+          { name: 'Georgia', abbrev: 'GEO' },
+          { name: 'Portugal', abbrev: 'POR' },
+          { name: 'Czechia', abbrev: 'CZE' },
+          { name: 'Turkey', abbrev: 'TUR' }
+        ]
+      },
+    }
+    puts 'Creating the Euros...'
+    euros = Competition.find_or_create_by!(name: 'Euros 2024', start_date: Date.new(2024, 06, 14), end_date: Date.new(2024, 07, 14), api_id: 2018, api_code: 'EC')
+    puts '.. created the Euros'
+
+    puts 'Creating or finding first round...'
+    first_round = Round.find_or_create_by!(name: 'Group Stage', number: 1, competition: euros, api_name: 'GROUP_STAGE')
+    euros.update!(current_round: first_round)
+    Round.find_or_create_by!(name: 'Round of 16', number: 2, competition: euros, api_name: 'LAST_16')
+    Round.find_or_create_by!(name: 'Quarter-finals', number: 3, competition: euros, api_name: 'QUARTER_FINALS')
+    Round.find_or_create_by!(name: 'Semi-finals', number: 4, competition: euros, api_name: 'SEMI_FINALS')
+    # TODO: Doesn't look like the API has the 3rd place playoff
+    # Round.find_or_create_by!(name: 'Third Place', number: 5, competition: euros, api_name: '3PPO')
+    Round.find_or_create_by!(name: 'Final', number: 6, competition: euros, api_name: 'FINAL')
+
+    puts 'Creating or finding groups...'
```

Comment:
My rake would abort when creating the groups with this error:
``` rake aborted!
ActiveRecord::RecordInvalid: Validation failed: Abbrev has already been taken
/Users/hivelocity/Documents/GitHub/predictor-api/lib/tasks/euros.rake:86:in `block (4 levels) in <main>'
/Users/hivelocity/Documents/GitHub/predictor-api/lib/tasks/euros.rake:84:in `each'
/Users/hivelocity/Documents/GitHub/predictor-api/lib/tasks/euros.rake:84:in `block (3 levels) in <main>'
/Users/hivelocity/Documents/GitHub/predictor-api/lib/tasks/euros.rake:81:in `each_key'
/Users/hivelocity/Documents/GitHub/predictor-api/lib/tasks/euros.rake:81:in `block (2 levels) in <main>'
<internal:/Users/hivelocity/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
<internal:/Users/hivelocity/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
-e:1:in `<main>'
Tasks: TOP => euros:create
(See full trace by running task with --trace)



This refactored code seems to have worked through:

 puts 'Creating or finding groups...'
    groups.each_key do |group_name|
      puts "...#{group_name}..."
      group_data = groups[group_name]
      group = Group.find_or_initialize_by(name: group_name, round: first_round)
      group.api_id = group_data[:api_id]
      group.api_code = group_data[:api_code]
      group.save!

      group_data[:teams].each do |team_hash|
        puts "Name: #{team_hash[:name]}, Abbrev: #{team_hash[:abbrev]}"
        team = Team.find_or_initialize_by(abbrev: team_hash[:abbrev])
        team.name = team_hash[:name]
        team.save!

        Affiliation.find_or_create_by!(team: team, group: group)
      end
    end

Reply:
Ah interesting. Did you have the DB from when we did this last? 
I'm assuming it's because you had a Team already in your DB that was using the an `abbrev` but the name was different.
For example, Czech Republic is Czechia from the API, but surely the same abbrev `CZE`

Reply:
hmm, I don't think so, could be wrong tho. I just yesterday brewed postgress, setting up the repos again (last time I coded for this app was on the old laptop I think 😅). GPT said something like, `team.save!`  would make sure of uniqueness. I also manually changed all of the last group's abbrv to TEST1, TEST2, TEST3 etc... same issue.